### PR TITLE
add features

### PR DESCRIPTION
**Pull request by lgor89**

### CODE REVIEW

File: app/controllers/webhooks_controller.rb [hunk 1]
```diff
@@ -1,5 +1,17 @@
 require 'net/http'
 class WebhooksController < ApplicationController
-	
+	skip_before_action :verify_authenticity_token
+	def callback
+	@bot_token = Rails.application.secrets.telegram_token
```
Comment (anatoliliotych) [inline app/controllers/webhooks_controller.rb]: style guide!

File: config/initializers/bot.rb [hunk 2]
```diff
@@ -0,0 +1,22 @@
+require 'net/http'
+require 'json'
+
+	@bot_token = Rails.application.secrets.telegram_token
+	@common_request = "https://api.telegram.org/bot#{@bot_token}/"
+	@url = "https://arcane-savannah-59671.herokuapp.com/"
```
Comment (anatoliliotych) [inline config/initializers/bot.rb]: to constant!

**Review by anatoliliotych — CHANGES_REQUESTED** [review]

File: app/controllers/TEST [hunk 3]
```diff
@@ -0,0 +1,12 @@
+  1. Move the cursor to this line.
```
Comment (anatoliliotych) [inline app/controllers/TEST]: what this file is doing here?

File: app/controllers/webhooks_controller.rb [hunk 4]
```diff
@@ -1,5 +1,16 @@
 require 'net/http'
 class WebhooksController < ApplicationController
-	
-	
+  skip_before_action :verify_authenticity_token
+  def callback
+    @bot_token = Rails.application.secrets.telegram_token
+    @response = params[:message][:text]
+    @common_request = "https://api.telegram.org/bot#{@bot_token}/"
```
Comment (anatoliliotych) [inline app/controllers/webhooks_controller.rb]: move url to constant

File: app/controllers/webhooks_controller.rb [hunk 5]
```diff
@@ -1,5 +1,16 @@
...
+    @chat_id = params[:message][:chat][:id]
+    if @response.casecmp('start').zero?
+      uri = URI(@common_request + 'sendMessage' + '?chat_id=' +
```
Comment (anatoliliotych) [inline app/controllers/webhooks_controller.rb]: please make this more readable

File: config/initializers/bot.rb [hunk 6]
```diff
@@ -0,0 +1,19 @@
...
+BOT_TOKEN = Rails.application.secrets.telegram_token
+COMMON_REQUEST = "https://api.telegram.org/bot#{BOT_TOKEN}/".freeze
+URL = 'https://arcane-savannah-59671.herokuapp.com/'.freeze
+def set_webhook
+  uri = URI(COMMON_REQUEST + 'setWebhook' + '?url=' + URL.to_s)
+  @res = Net::HTTP.get(uri)
+  p @result = JSON.parse(@res)
+end
+
+def get_me
```
Comment (anatoliliotych) [inline config/initializers/bot.rb]: check naming for the method

**Review by anatoliliotych — CHANGES_REQUESTED** [review]

File: app/controllers/TEST (hunk 3 above)
Comment (lgor89) [reply app/controllers/TEST]: I don't know. Maybe this is Rails magic)

File: database.yml [hunk 7]
```diff
@@ -0,0 +1,2 @@
+./
```
Comment (anatoliliotych) [inline database.yml]: strange file...

**Review by anatoliliotych — CHANGES_REQUESTED** [review]
This bot with bot gem will take couple lines of code... 
I shouldn't found any gem, wich make updates through webhooks, so wrote myself )

File: database.yml (hunk 7 above)
Comment (lgor89) [reply database.yml]: where you found it?

File: app/controllers/webhooks_controller.rb [hunk 8]
```diff
@@ -1,5 +1,15 @@
...
 class WebhooksController < ApplicationController
-	
-	
+  skip_before_action :verify_authenticity_token
+  COMMON_REQUEST = Rails.application.secret.common_request
+  def callback
+    @response = params[:message][:text]
+    @chat_id = params[:message][:chat][:id]
+    @name = params[:message][:from][:first_name]).to_s)
+    if @response.casecmp('start').zero?
```
Comment (anatoliliotych) [inline app/controllers/webhooks_controller.rb]: not the best way to check if the substring is part of the string

File: app/controllers/webhooks_controller.rb [hunk 9]
```diff
@@ -1,5 +1,15 @@
...
+      parameters = URI.encode_www_form("chat_id"=>@chat_id.to_s,"text"=>"hello #{@name}")
```
Comment (anatoliliotych) [inline app/controllers/webhooks_controller.rb]: check style guide

File: app/controllers/webhooks_controller.rb [hunk 10]
```diff
@@ -1,5 +1,15 @@
...
+      uri = URI(COMMON_REQUEST + 'sendMessage?'+"#{parameters}")
```
Comment (anatoliliotych) [inline app/controllers/webhooks_controller.rb]: check style guide

**Review by anatoliliotych — CHANGES_REQUESTED** [review]
Please add readme

**Pull request closed without merging**